lib/SyTest/Homeserver.pm

```diff
@@ -82,14 +82,25 @@ sub clear_db_pg
    require DBI;
    require DBD::Pg;
 
-   my $dbh = DBI->connect( "dbi:Pg:dbname=$args{database};host=$host", $args{user}, $args{password} )
-      or die DBI->errstr;
+   # If there is a DB called sytest_template use that as the template for the
+   # sytest databases. Otherwise initialise the DB from scratch (which can take
+   # a fair few seconds)
+   my $dbh = DBI->connect( "dbi:Pg:dbname=sytest_template;host=$host", $args{user}, $args{password} );
+   if ( $dbh ) {
+      $dbh->do( "DROP DATABASE $args{database}" );  # we don't mind if this dies
+
+      $dbh->do( "CREATE DATABASE $args{database} WITH TEMPLATE sytest_template" ) or
+         die $dbh->errstr;
+   } else {
+      my $dbh = DBI->connect( "dbi:Pg:dbname=$args{database};host=$host", $args{user}, $args{password} )
```

Comment:
You probably didn't want to make a new `$dbh` in the body of this `else` block, or else it won't be visible to code below. I suggest remove the `my`.